### PR TITLE
Feature/issue 741 - Remote Control Radio Parameter Update

### LINKED ISSUE #741: [SDL 0160] Remote Control Radio Parameter Update
Proposal: [Remote Control Radio Parameter Update](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0160-rc-radio-parameter-update.md)

> This proposal adds a new parameter `hdRadioEnable` to `RadioControlData` data structure, adds new parameters `hdRadioEnableAvailable` and `siriusxmRadioAvailable` to `RadioControlCapabilities` data structure, and changes the maximum number of HD radio sub-channels from 3 to 7.

Review: https://github.com/smartdevicelink/sdl_evolution/issues/464

### PR DESCRIPTION
**Pull request by mrapitis**
Fixes #741 

This PR is **ready** for review.

## Risk

This PR makes **minor** API changes.

## Testing Plan

Added new unit tests and expanded existing ones

## Summary

Added RPC's, structs, enums as defined in the Remote Control Radio Parameter Update proposal.

## CLA

- [x] I have signed the CLA

### CODE REVIEW

File: sdl_android/src/androidTest/java/com/smartdevicelink/test/rpc/datatypes/RadioControlCapabilitiesTests.java [hunk 1]
```diff
@@ -12,101 +12,111 @@
 import java.util.Iterator;
 
 /**
- * This is a unit test class for the SmartDeviceLink library project class : 
+ * This is a unit test class for the SmartDeviceLink library project class :
```
Comment (theRealBrett) [inline sdl_android/src/androidTest/java/com/smartdevicelink/test/rpc/datatypes/RadioControlCapabilitiesTests.java]: Is it possible to only add the new additions?

File: sdl_android/src/androidTest/java/com/smartdevicelink/test/rpc/datatypes/RadioControlDataTests.java [hunk 2]
```diff
@@ -18,107 +18,112 @@
 import java.util.Iterator;
 
 /**
- * This is a unit test class for the SmartDeviceLink library project class : 
+ * This is a unit test class for the SmartDeviceLink library project class :
```
Comment (theRealBrett) [inline sdl_android/src/androidTest/java/com/smartdevicelink/test/rpc/datatypes/RadioControlDataTests.java]: same with this class, is it possible to only show the additions?

File: sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/RadioControlCapabilities.java [hunk 3]
```diff
@@ -3,246 +3,280 @@
 import android.support.annotation.NonNull;
 
 import com.smartdevicelink.proxy.RPCStruct;
+
 import java.util.Hashtable;
 
 /**
  * Contains information about a radio control module's capabilities.
  */
-public class RadioControlCapabilities extends RPCStruct{
```
Comment (theRealBrett) [inline sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/RadioControlCapabilities.java]: is it possible to only add the new code?

File: sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/RadioControlData.java [hunk 4]
```diff
@@ -3,221 +3,231 @@
...
 import com.smartdevicelink.proxy.rpc.enums.RadioBand;
 import com.smartdevicelink.proxy.rpc.enums.RadioState;
+
 import java.util.Hashtable;
 
 /**
  * Include information (both read-only and changeable data) about a
  * remote control radio module.
  */
-public class RadioControlData extends RPCStruct{
```
Comment (theRealBrett) [inline sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/RadioControlData.java]: Can you please only add the new code to the class?

**Comment by mrapitis** [comment]
@theRealBrett PR has been updated per review comments, please re-review when time permits.  Thanks!

**Comment by joeygrover** [comment]
@mrapitis changes look good, but can you pull in develop and resolve conflicts?

**Comment by mrapitis** [comment]
Sure thing, we will resolve the conflicts and update when ready.

**Comment by mrapitis** [comment]
@joeygrover we have pulled in the current develop branch, please re-review when time permits.  Thanks!

**Pull request merged**